UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs

```diff
@@ -205,6 +205,22 @@ protected virtual void Awake()
 		IPlayerEntersTiles = GetComponents<IPlayerEntersTile>();
 		IObjectEntersTiles = GetComponents<IObjectEntersTile>();
 		CurrentsortingGroup = GetComponent<SortingGroup>();
+
+		if (GameManager.Is3D)
+		{
+			var Is3D = this.gameObject.GetComponent<ConvertTo3D>();
+			if (Is3D == null)
+			{
+
+				Is3D =this.gameObject.AddComponent<ConvertTo3D>();
+
+			}
+			Is3D?.DoConvertTo3D();
```

Comment:
elvis operator shouldn't be needed here since you are making sure you are adding the component.